libraries/AP_Rally/AP_Rally.cpp

```diff
@@ -161,3 +171,40 @@ Location AP_Rally::calc_best_rally_or_home_location(const Location &current_loc,
 
     return return_loc;
 }
+// return best rally location from current position - not including home or km limit
+Location AP_Rally::calc_best_rally_location(const Location &current_loc, float rtl_home_alt) const
+{
+    RallyLocation ral_loc = {};
+    Location return_loc = {};
+    const struct Location &home_loc = _ahrs.get_home();
+    
+    find_nearest_rally_point(current_loc, ral_loc);  
+    return_loc = rally_location_to_location(ral_loc);     
+    if (!is_valid(return_loc)) {
+        // no valid rally point, return home position
+        return_loc = home_loc;
+        return_loc.alt = rtl_home_alt;
+        return_loc.flags.relative_alt = false; // read_alt_to_hold returns an absolute altitude
+    }
+
+    return return_loc;
+}
+
+//return location of a specific id rally point
+Location AP_Rally::go_to_rally_id_location(const Location &current_loc, uint8_t rally_id) const
```

Comment:
I don't see any reason for this method, the caller can already get the specific rally point and if doesn't exist it can then choose what to do.

Reply:
Agree, the `const Location &current_loc` is too much, it was a bad copy paste from other method.  However, I think the rest of the function is fine.
Maybe I can rename to `get_rally_location_with_index` to have better consistency with `get_rally_point_with_index`

Reply:
What is this needed for? I see no reason to be hard-coding the home behavior when the caller can do the same - if this was going to be repeated in a lot of places then it would make sense but this is going to be used in one place.

Reply:
Ok, I can do that. I thought it was cleaner to define an alternate method even though it's called once (for now?).